flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveDialectITCase.java

```diff
@@ -539,6 +539,35 @@ public void testFunction() throws Exception {
         tableEnv.executeSql("drop function if exists foo");
     }
 
+    @Test
+    public void testTemporaryFunction() throws Exception {
+        // create temp function
+        tableEnv.executeSql(
+                String.format(
+                        "create temporary function temp_abs as '%s'",
+                        GenericUDFAbs.class.getName()));
+        List<Row> functions =
+                CollectionUtil.iteratorToList(tableEnv.executeSql("show functions").collect());
+        assertTrue(functions.toString().contains("temp_abs"));
```

Comment:
We can now use `tableEnv.listUserDefinedFunctions()` or `tableEnv.executeSql("show user functions")` to verify exactly the user defined functions.